Check against 10701 again

adamruzicka opened the pull request:
Successor to https://github.com/Katello/katello/pull/11512, I could have sworn the whole suite was green before.

## Summary by Sourcery

CI:
- Set foreman_version to pull/10701/head in the Ruby workflow

## Summary by Sourcery

Update tests to use proper proxy context and enhance permission checks, and point CI to Foreman pull request head.

Build:
- Set foreman_version to refs/pull/10701/head in the Ruby GitHub Actions workflow

Tests:
- Refactor CapsuleContentController tests to use @proxy with associated organization and location contexts
- Add organization and location parameters to protected action tests for CapsuleContentController
- Include organization in scoped_search permissions for activation key tests
- Add organization context to protected tests in HostCollectionsController and SyncPlansController

adamruzicka committed:
Refs #38844 - Adjust tests to foreman#10701, continued

ianballou reviewed (commented):
ACK, we can merge once the foreman PR is pulled out.

adamruzicka committed:
Fix capsule_content_test

Previously, the Smart Proxy being passed around didn't belong to any
organizations, so a call to verify_ueber_certs, which iterates over the
proxy's organizations was essentially a noop.
